### DISCUSSION TITLE
Impossible to connect "Selected peer config '...' unacceptable: constraint checking failed

### DISCUSSION THREAD

**Discussion by SAOIRI123** [strongswan/strongswan#903] [Q&A] [answered]
Hi everyone,

I have a GenIP 30i module on which I'm trying to connect in IKEv2. I don't have access to many parameters on this module.
Here is what I can set:
![image](https://user-images.githubusercontent.com/97451715/154275833-b5aae2ec-a9a1-4951-b561-8dac257bcb31.png)

but when I try to connect I get the following message:
  Feb 16 14:32:11 sdeau charon-nm: 09[IKE] authentication of '****' (myself) with pre-shared key
  Feb 16 14:32:11 sdeau charon-nm: 09[IKE] establishing CHILD_SA Auchan_GenIP_30i{1}
  Feb 16 14:32:11 sdeau charon-nm: 09[ENC] generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) CERTREQ AUTH SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
  Feb 16 14:32:11 sdeau charon-nm: 09[NET] sending packet: from ***[40718] to ***[4500] (5532 bytes)
  Feb 16 14:32:15 sdeau charon-nm: 12[IKE] retransmit 1 of request with message ID 1
  Feb 16 14:32:15 sdeau charon-nm: 12[NET] sending packet: from ***[40718] to ***[4500] (5532 bytes)
  Feb 16 14:32:15 sdeau charon-nm: 13[NET] received packet: from ***[4500] to ***[40718] (220 bytes)
  Feb 16 14:32:15 sdeau charon-nm: 13[ENC] parsed IKE_AUTH response 1 [ IDr AUTH SA TSi TSr N(AUTH_LFT) ]
  Feb 16 14:32:15 sdeau charon-nm: 13[IKE] authentication of '***' with pre-shared key successful
  Feb 16 14:32:15 sdeau charon-nm: 13[CFG] constraint check failed: identity '***' required 
  Feb 16 14:32:15 sdeau charon-nm: 13[CFG] selected peer config 'Auchan_GenIP_30i' unacceptable: constraint checking failed
  Feb 16 14:32:15 sdeau charon-nm: 13[CFG] no alternative config found
  Feb 16 14:32:15 sdeau charon-nm: 13[ENC] generating INFORMATIONAL request 2 [ N(AUTH_FAILED) ]

On ubuntu:
![image](https://user-images.githubusercontent.com/97451715/154276507-7342a283-5ab8-4e05-bb17-82f4c3ab6f41.png)

Any ideas?
Thanks

**Comment by tobiasbrunner** [comment]
Might be tricky. What other VPN types does this device support? Because "Automatic subnet to subnet" isn't what the NM client is designed for (it's more for roadwarrior connections). For instance, you can't specify the proposed traffic selectors. If you can't change it to something more appropriate, it might be better to configure strongSwan via config file and not use the NM plugin.

Also, the issue you are currently having is an identity mismatch. Since the NM plugin you are using is older, it doesn't provide an option to set the server identity explicitly, so the configured server address is used. This has to match whatever the server sends as its identity. I don't see an option there to change it, so again you might have more success with configuring this via config file as there you can set the server identity explicitly.

**Comment by SAOIRI123** [comment]
first of all, thank you for your answer.

I'm afraid I don't have the skills required for the configuration. All the examples on the internet use way too many parameters...

I don't even know where to start.

**Reply by tobiasbrunner** [reply]
You might want to start with [UsableExamples](https://wiki.strongswan.org/projects/strongswan/wiki/UsableExamples). Which of the configs applies depends on your actual use case.

**Reply by SAOIRI123** [reply]
Yes, that's what I've been trying to make work for the last 3 or 4 days before I gave up. In addition to this [tuto](https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-20-04-fr) But I have way too many files and certificates while the GenIP 30i which is the server doesn't require any of that.

**Reply by tobiasbrunner** [reply]
If you use a pre-shared key (PSK) to authenticate, you don't need any certificates and only have to configure a single file, swanctl.conf (or two if you use ipsec.conf/ipsec.secrets). It's really quite simple, you need to specify the remote address/hostname, the local and remote identity (unless they equal the IP addresses), set the authentication to PSK and specify the local and remote subnets (in CIDR notation). Then configure the PSK for the remote identity and you are done. Everything else is just on top of this basic config.

**Comment by SAOIRI123** [comment, marked as answer]
Hello,

I had access to the file on the genip debug page, here is what we can see:

cat /etc/ipsec.conf
conn %default
ikelifetime=6h
keylife=6h
rekeymargin=9m
keyingtries=%forever
authby=secret
mobike=no
dpddelay=60s
dpdtimeout=180s
dpdaction=none
rightallowany=yes
inactivity=0

conn tunnel_0_net_net
leftupdown="ipsec _updown;/home/root/applications/erco/command_logger ADD_IPSEC_UPDOWN INFO tunnel_0_net_net/Auchan_GenIP_30i"
type=tunnel
keyexchange=ikev2
installpolicy=yes
ike=aes128-sha1-modp2048!
esp=aes128-sha1-modp2048,aes128-sha1-modp2048,3des-sha1-modp1536!
compress=no
left=%defaultroute
leftikeport=500
leftid=LEFTID
lefthostaccess=yes
right=%any
rightikeport=500
rightid=RIGHTID
righthostaccess=yes
rightfirewall=yes
leftsubnet=192.168.1.0/24
rightsubnet=10.0.2.0/8
auto=add
closeaction=restart
dpddelay=60s
dpdtimeout=180s
dpdaction=none




cat /etc/ipsec.secrets
LEFTID RIGHTID : PSK "*********"


I tried to copy and paste the configuration replacing left by right and right by left. but when I do: sudo systemctl start strongswan-starter. Nothing happens (not even an error message).

Do you have an idea?

thanks

Edit sorry guys i fill like really silly...

**Reply by tobiasbrunner** [reply]
> replacing left by right and right by left

Note that some options only make sense with `left` prefix e.g. `leftfirewall` or `leftupdown` (although you probably want to remove the latter anyway). And `right` has to be set to the hostname/IP of the peer, using `%defaultroute` makes no sense. `left|rightid` and `left|rightsubnet` you can switch.

> but when I do: sudo systemctl start strongswan-starter. Nothing happens (not even an error message).

You can check the status (e.g. if the connection is loaded) with `ipsec statusall` (you can also check the log for details) and for connections with `auto=add` you have to manually initiate them with `ipsec up` (see [Invocaton and Maintenance](https://wiki.strongswan.org/projects/strongswan/wiki/IntroductionTostrongSwan#Invocation-and-Maintenance)).

**Reply by SAOIRI123** [reply]
If I understand correctly, it should look something like this (on ubuntu the client):
conn tunnel_0_net_net
    type=tunnel
    keyexchange=ikev2
    installpolicy=yes
    ike=aes128-sha1-modp2048!
    esp=aes128-sha1-modp2048,aes128-sha1-modp2048,3des-sha1-modp1536!
    compress=no
    leftikeport=500
    leftid=RIGHTID
    right=9****
    rightikeport=500
    rightid=LEFTID
    rightsubnet=192.168.1.0/24
    leftsubnet=10.0.2.0/8
    auto=add

this is my ipsec.secret:
LEFTID RIGHTID : PSK "PSK-TEST"

with "sudo ipsec up tunnel_0_net_net" I get the following message:
initiating IKE_SA tunnel_0_net_net[3] to 9****
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.0.2.15[500] to 9***[500] (464 bytes)
received packet: from 9***[500] to 10.0.2.15[500] (456 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(MULT_AUTH) ]
selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_2048
local host is behind NAT, sending keep alives
no private key found for 'RIGHTID'

really sorry, you must hate me a lot.

**Reply by tobiasbrunner** [reply]
Hm, `leftid` should be set to your identity and `rightid` to the other's. And to use a pre-shared key to authenticate, instead of certificate/public key authentication, you have to configure `authby=psk`.

**Reply by SAOIRI123** [reply]
IT'S WORK !!! Thanks 👍 thanks, thanks
